web/src/compose.js

```diff
@@ -338,6 +338,7 @@ export function finish() {
         send_message();
     }
     do_post_send_tasks();
+    compose_ui.hide_compose_spinner();
```

Comment:
This will hide the compose box spinner too early when not locally echoing -- I think you want to put this within this block inside `send_message` ... and maybe in a similar place inside `schedule_message`
```
    function error(response) {                                                                          
        // If we're not local echo'ing messages, or if this message was not                             
        // locally echoed, show error in compose box                                                    
        if (!locally_echoed) {                                                                          
            compose_banner.show_error_message(                                                          
                response,                                                                               
                compose_banner.CLASSNAMES.generic_compose_error,                                        
                $("#compose-textarea"),                                                                 
            );                                                                                          
            return;                                                                                     
        }                                                                                               
```